.github/workflows/ci.yml

```diff
@@ -96,25 +77,27 @@ jobs:
       - name: Install Prettier
         run: npm install --global prettier
       - name: Install Nightly Toolchain
-        uses: actions-rs/toolchain@v1
+        uses: actions-rust-lang/setup-rust-toolchain@v1
         with:
           toolchain: nightly
           components: rustfmt, clippy
-      - name: Cache Rust Dependencies
-        uses: actions/cache@v4
-        with:
-          path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-            target
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
       - uses: actions/checkout@v4
+        with:
+          ref: ${{ (env.LINT_MODE == 'fix' && github.event.pull_request.head.sha) || '' }}
       - name: Run Lint Script
-        run: ./lint.sh --mode=check
+        run: ./lint.sh --mode=$LINT_MODE
+      - name: Commit and push if changed
+        if: env.LINT_MODE == 'fix'
+        # TODO: works only for internal contributors
+        # to make it for forks we need smth like https://autofix.ci
+        continue-on-error: true
+        uses: stefanzweifel/git-auto-commit-action@v5
```

Comment:
so without `pull_request_target` it won't work on forks.

As the solution we can install this [app](https://autofix.ci) in our repo. It's free for open-source repos